### PR TITLE
Restart button fix

### PR DESCRIPTION
**Pull request by dydzio0614**
Added bool variable that prevents game from starting if server is running. That guarantees server to close in time.

### CODE REVIEW

File: client/CMT.cpp [hunk 1]
```diff
@@ -90,6 +90,7 @@ std::queue<SDL_Event> events;
 boost::mutex eventsM;
 
 bool gNoGUI = false;
+bool serverNotDead = false;
```
Comment (alexvins) [inline client/CMT.cpp]: Ensure synchronized access to this variable. (smth from std::atomic f.e.)

File: client/CMT.cpp [hunk 2]
```diff
@@ -1248,6 +1249,10 @@ static void mainLoop()
 
 void startGame(StartInfo * options, CConnection *serv/* = nullptr*/)
 {
+	while (serverNotDead == true)
```
Comment (alexvins) [inline client/CMT.cpp]: write just `while (serverNotDead)`

File: client/CMT.cpp (hunk 1 above)
Comment (alexvins) [inline client/CMT.cpp]: May be serverAlive is better name.

Comment (dydzio0614) [inline client/CMT.cpp]: What is the reason though?

**Comment by mwu-tow** [comment]
Check the `CondSh<bool>`  — it serves this exact idiom (flag with synchronized access that can be waited upon). With that you can get rid of those ugly sleep calls.

**Comment by dydzio0614** [comment]
Will do

**Pull request merged**